src/main/java/goral/psychotherapistoffice/config/security/CustomSecurityConfig.java

```diff
@@ -20,9 +20,10 @@ public class CustomSecurityConfig {
     @Bean
     public SecurityFilterChain filterChain(HttpSecurity httpSecurity) throws Exception{
         httpSecurity.authorizeHttpRequests((authz) -> authz
+                        .requestMatchers("/termin/**", "/user/**").hasAnyRole(USER_ROLE, ADMIN_ROLE)
```

Comment:
have you role hierarchy in java?

for example in my app for Świdnik i can define hierarchy in symfony. and my `ROLE_ADMIN` has `ROLE_EMPLOYEE`, and  `ROLE_EMPLOYEE` has role `ROLE_USER`. 

for this function `.hasAnyRole(USER_ROLE)`  its working for all my roles

my security.yaml
```
security:
    role_hierarchy:
        ROLE_EMPLOYEE: ROLE_USER
        ROLE_ADMIN: ROLE_EMPLOYEE
        ROLE_DIRECTOR: [ ROLE_ADMIN, ROLE_ALLOWED_TO_SWITCH, ROLE_SHOW_TASKS ]
        ROLE_SUPER_ADMIN: [ ROLE_ADMIN, ROLE_ALLOWED_TO_SWITCH, ROLE_DIRECTOR, ROLE_DEBUG_SCHEDULE, ROLE_SHOW_TASKS ]
```

Reply:
2. mysql dependency stayed because I unsuccessfully tried to configure the production profile in my API
3. I recommend Eclipse is free and many people write in Eclipse I do too
https://javastart.pl/baza-wiedzy/wpowiedzi/eclipse-ide-wstep-i-instalacja
Or If you prefer visual studio code
4 I know the topic of Odi. I am sad because he died. Even I was sad and it was difficult for you. But my proposition about Bambi cat is still actually.
5. Nice decision, congratulations, we will talk about it today, because I will come to Krasnystaw to pick up Wojtek
6. The band will accompany me in coding next to TonyHowk's playlist :) Thank you!

Reply:
have you role hierarchy in java?

I'm understanding this approach.It looks nice.
I tried to configure yaml file and is no possible in similar way as in Świdnik.
I watched Spring tutorials and they use spring security dependency, they configure request matcher as I javastart course It is probably the easiest way for beginners so I accepted this:)
but sometimes looking for users repositories so I will be remember to compare code in this area.
Thank you for your suggestions.

Reply:
src/main/java/goral/psychotherapistoffice/config/security/CustomSecurityConfig.java